src/run/headers.ts

```diff
@@ -208,6 +209,19 @@ export const adjustDateHeader = async ({
   headers.set('date', lastModifiedDate.toUTCString())
 }
 
+function setCacheControlFromRequestContext(
+  headers: Headers,
+  revalidate: NetlifyCachedRouteValue['revalidate'],
+) {
+  const cdnCacheControl =
+    // if we are serving already stale response, instruct edge to not attempt to cache that response
+    headers.get('x-nextjs-cache') === 'STALE'
+      ? 'public, max-age=0, must-revalidate, durable'
+      : `s-maxage=${revalidate || 31536000}, stale-while-revalidate=31536000, durable`
```

Comment:
I think it should be okay to use `${revalidate || 31536000}` because we need to account for `false` and `undefined` (see comment below), and even if `revalidate` was `0` then we should be using 1 year.